aten/src/ATen/core/type.cpp

```diff
@@ -585,6 +585,22 @@ bool is_null_or_equal(c10::optional<T> a, c10::IntArrayRef b) {
   return !a.has_value() || a.value() == b;
 }
 
+
+
+ bool matchStrides(TensorType* ttp, const at::Tensor& t) {
+   TORCH_INTERNAL_ASSERT(t.is_contiguous());
+   auto ssize = *ttp->stride_properties().size();
+   const auto& t_strides = t.strides();
+   for (size_t i = 0; i < ssize; i++) {
+     const auto& stride = *ttp->stride_properties()[i];
```

Comment:
similarly, we're also assuming every stride property is set, which isn't generally valid